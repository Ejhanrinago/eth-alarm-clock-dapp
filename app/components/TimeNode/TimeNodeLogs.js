import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { inject, observer } from 'mobx-react';
import { LOGGER_MSG_TYPES } from '../../lib/worker-logger.js';

@inject('timeNodeStore')
@observer
class TimeNodeLogs extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showCache: true,
      showInfo: true,
      showDebug: true,
      showError: true
    };
  }

  updateFilter(type) {
    if (type === LOGGER_MSG_TYPES.CACHE) {
      this.setState({
        showCache: event.target.checked
      });
    }
  }

  formatUnix(unix) {
    return moment.unix(unix).format('DD/MM/YYYY HH:mm:ss');
  }

  render() {
    return (
      <div id="timeNodeLogs">
        <div className="row">

          <div className="col-9 col-sm-9 col-md-10">
            <div data-pages="card" className="card card-default">
              <div className="card-header">
                <div className="card-title">Logs</div>
                <div className="card-controls">
                  <ul>
                    <li>
                      <a data-toggle="refresh" className="card-refresh" href="#"><i className="card-icon card-icon-refresh"></i></a>
                    </li>
                  </ul>
                </div>
              </div>
              <div className="card-body">
                {this.props.timeNodeStore.logs.map(([ timestamp, message, type ], index) => {
                  return (
                    <p key={index} className="no-margin">
                      <span>{this.formatUnix(timestamp)}</span> [{type.toUpperCase()}] {message}
                    </p>
                  );
                })}
              </div>
            </div>
          </div>

          <div className="col-3 col-sm-3 col-md-2">
            <p><b>Filter:</b></p>

            <div className="checkbox check-primary">
              <input type="checkbox"
                defaultChecked={this.state.showCache}
                id="checkboxCache"
                onChange={() => this.updateFilter(LOGGER_MSG_TYPES.CACHE)}/>
              <label htmlFor="checkboxCache">{LOGGER_MSG_TYPES.CACHE}</label>
            </div>

            <div className="checkbox check-info">
              <input type="checkbox" defaultChecked={this.state.showInfo} id="checkboxInfo"/>
              <label htmlFor="checkboxInfo">{LOGGER_MSG_TYPES.INFO}</label>
            </div>

            <div className="checkbox check-warning">
              <input type="checkbox" defaultChecked={this.state.showDebug} id="checkboxDebug"/>
              <label htmlFor="checkboxDebug">{LOGGER_MSG_TYPES.DEBUG}</label>
            </div>

            <div className="checkbox check-danger">
              <input type="checkbox" defaultChecked={this.state.showError} id="checkboxError"/>
              <label htmlFor="checkboxError">{LOGGER_MSG_TYPES.ERROR}</label>
            </div>
          </div>

        </div>
      </div>
    );
  }
}

TimeNodeLogs.propTypes = {
  timeNodeStore: PropTypes.any
};

export default TimeNodeLogs;
