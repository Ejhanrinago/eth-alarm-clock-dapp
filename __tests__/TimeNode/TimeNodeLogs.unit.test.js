import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import TimeNodeLogs from '../../app/components/TimeNode/TimeNodeLogs';

describe('TimeNodeLogs', () => {
  it('correctly displays logs', () => {
    const timeNodeStore = {
      logs: [
        {
          timestamp: '1521032727',
          type: 'DEBUG',
          message: 'Scanning started.'
        },
        {
          timestamp: '1521032730',
          type: 'INFO',
          message: 'Transaction found.'
        }
      ]
    };

    const injectables = {
      timeNodeStore
    };

    let mockedRender = renderer.create(
      <Provider {...injectables}>
        <TimeNodeLogs />
      </Provider>,
    );

    let tree = mockedRender.toJSON();
    expect(tree).toMatchSnapshot();
  });
});