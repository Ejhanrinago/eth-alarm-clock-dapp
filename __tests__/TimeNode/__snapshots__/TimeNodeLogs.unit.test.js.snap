// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeNodeLogs correctly displays logs 1`] = `
<div
  id="timeNodeLogs"
>
  <div
    className="m-b-20"
  >
    <div
      className="btn-group"
      data-toggle="buttons"
    >
      <label
        aria-pressed="true"
        className="btn btn-default active"
        data-log-type="basic"
        onClick={[Function]}
      >
        <input
          name="options"
          type="radio"
        />
        Basic
      </label>
      <label
        aria-pressed="true"
        className="btn btn-default"
        data-log-type="detailed"
        onClick={[Function]}
      >
        <input
          checked=""
          name="options"
          type="radio"
        />
        Detailed
      </label>
    </div>
  </div>
  <div
    className="card card-default"
    data-pages="card"
  >
    <div
      className="card-header"
    >
      <div
        className="card-title"
      >
        Logs
      </div>
      <div
        className="card-controls"
      >
        <ul>
          <li>
            <a
              className="card-refresh"
              data-toggle="refresh"
              href="#"
            >
              <i
                className="card-icon card-icon-refresh"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="card-body"
      id="timenodeLogsField"
    >
      <p
        className="no-margin"
      >
        <span>
          14/03/2018 14:05:27
        </span>
         [
        DEBUG
        ] 
        Scanning started.
      </p>
      <p
        className="no-margin"
      >
        <span>
          14/03/2018 14:05:30
        </span>
         [
        INFO
        ] 
        Transaction found.
      </p>
    </div>
  </div>
</div>
`;
